physical/raft/fsm.go

```diff
@@ -784,7 +784,15 @@ func (f *FSM) Restore(r io.ReadCloser) error {
 
 	snapshotInstaller, ok := r.(*boltSnapshotInstaller)
 	if !ok {
-		return errors.New("expected snapshot installer object")
+		wrapper, ok := r.(raft.ReadCloserWrapper)
+		if !ok {
+			return fmt.Errorf("expected ReadCloserWrapper object, got: %T", r)
+		}
+		snapshotInstallerRaw := wrapper.WrappedReadCloser()
+		snapshotInstaller, ok = snapshotInstallerRaw.(*boltSnapshotInstaller)
+		if !ok {
+			return fmt.Errorf("expected snapshot installer object, got: %T", r)
```

Comment:
Do you want to pass `r` here to `fmt.Errorf()` or should this be `snapshotInstallerRaw`, since that's what failed the type coercion on line 792?

Reply:
Good catch, ty.

Reply:
Fixed.